gramex/handlers/mlhandler.py

```diff
@@ -275,14 +291,18 @@ def _train(self, data=None):
         target_col = self.get_argument('target_col', self.store.load('target_col'))
         index_col = self.get_argument('index_col', self.store.load('index_col'))
         self.store.dump('target_col', target_col)
-        data = self._parse_data(False) if data is None else data
-        data = self._filtercols(data)
-        data = self._filterrows(data)
-        self.model = get_model(
-            self.store.load('class'), self.store.load('params'),
-            data=data, target_col=target_col,
-            nums=self.store.load('nums'), cats=self.store.load('cats')
-        )
+        if isinstance(self.model, ml.KerasApplication):
+            result = self.model.fit(data, self.kwargs['config_dir'])
+            return result
+        else:
+            data = self._parse_data(False) if data is None else data
+            data = self._filtercols(data)
+            data = self._filterrows(data)
+            self.model = get_model(
+                self.store.load('class'), self.store.load('params'),
+                data=data, target_col=target_col,
+                nums=self.store.load('nums'), cats=self.store.load('cats')
+            )
```

Comment:
@jaidevd  - find a way to remove dataframe-specific code from MLHandler - it should deal only with train / predict semantics.